server/integration/infinispan/src/main/java/org/infinispan/server/infinispan/spi/service/CounterServiceNameFactory.java

```diff
@@ -0,0 +1,18 @@
+package org.infinispan.server.infinispan.spi.service;
+
+import org.jboss.msc.service.ServiceName;
+
+/**
+ * Factory for generating service names for services associated with a counter.
+ * @author Vladimir Blagojevic
+ */
+public interface CounterServiceNameFactory {
```

Comment:
I'm not sure but this interface isn't needed (IMO).
It is only implemented by enum `CounterServiceName.CONFIGURATION`

Reply:
It is, believe me! I did not want to use hacks in other service factories and mix apples and oranges. All it does it appends container name with counter name. And we need that for the uniqueness of counter. I could have hacked it with other service factories but it would be a kludge

Reply:
Why not just have: 
```java
public class CounterServiceName {
    public static ServiceName getServiceName(String container, String counter) {
        return CacheContainerServiceName.CACHE_CONTAINER.getServiceName(container).append(counterName)
    }
}
```
Or you could just add ^ method to `CounterService.java`

Either way there is no need for the interface.